### PR TITLE
Refactor git-ls-files into a common utils function

### LINKED ISSUE #15: refactor git-ls-files into a common utils function

### PR DESCRIPTION
**Pull request by abulenok**
closes #15

### CODE REVIEW

File: test_notebooks.py [hunk 1]
```diff
@@ -24,12 +24,15 @@
 SI = pint.UnitRegistry()
 
 
+def find_files(path_to_folder_from_project_root=".", file_extension=".ipynb"):
```
Comment (slayoo) [inline test_notebooks.py:27]: let's also use it in here: https://github.com/open-atmos/devops_tests/blob/269f8685a1973f8a788562ddbc918e6629017893/test_todos_annotated.py#L26

Comment (abulenok) [reply test_notebooks.py]: Added

**Comment by slayoo** [comment]
closing & reopening to trigger CI

**Comment by slayoo** [comment]
@abulenok both pylint and pre-commit complain here

File: test_notebooks.py [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 import pint
 import pytest
-from git.cmd import Git
+
+from utils import find_files
+
+if sys.platform == "win32" and sys.version_info[:2] >= (3, 7):
+    import asyncio
+
+    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
```
Comment (slayoo) [inline test_notebooks.py:18]: @abulenok moving it here will break this workaround. We should find a way of depicting this in devops-tests CI...

Comment (slayoo) [reply test_notebooks.py:18]: I think this is the relevant issue: https://github.com/jupyter/nbclient/issues/85

Comment (slayoo) [reply test_notebooks.py:18]: and likely it needs to be imported before `nbformat`?

Comment (slayoo) [reply test_notebooks.py:18]: (which is why it was placed above with the `pylint: disable=wrong-import-position` added)

File: test_notebooks.py [hunk 3]
```diff
@@ -15,7 +15,9 @@
 import nbformat
 import pint
 import pytest
-from git.cmd import Git
+
+from utils import find_files
```
Comment (slayoo) [inline test_notebooks.py:19]: perhaps a relative import, not to confuse with https://pypi.org/project/utils?

Comment (abulenok) [reply test_notebooks.py:19]: `attempted relative import with no known parent package`

Comment (slayoo) [reply test_notebooks.py:19]: so maybe a less general name, like `devops_tests_commons`?

Comment (abulenok) [reply test_notebooks.py:19]: maybe make a package from devops_tests?

Comment (slayoo) [reply test_notebooks.py:19]: the way we use it, through `pytest` which scans directories for `test*.py` files, would ignore anything pip-installed, so making it an installable package would likely not have any added value... but just adding an `__init__.py` in the root dir (without `setup.py` or `pyproject.toml`) sounds reasonable.

Comment (abulenok) [reply test_notebooks.py]: updated

File: utils.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+"""
+Utils functions to reuse in different parts of the codebase
+"""
+import os
+
+from git.cmd import Git
+
+
+def find_files(path_to_folder_from_project_root=".", file_extension=None):
```
Comment (slayoo) [inline utils.py:9]: OK, last comment :)
I'd propose adding a unit test for `find_files` - the point is that in case `find_files` starts to return an empty set of paths, all our existing CI tests will... succeed as they all iterate over files returned by `find_files`. So, perhaps let's just check if it finds `.git/config` or `README.md`?

Comment (abulenok) [reply utils.py:9]: or maybe an assert? assert that .git/config exists in the list. The test will not save us from incorrect use of find_files.

Comment (slayoo) [reply utils.py:9]: well, interestingly actually something has changed in this PR which makes the "selftest" job execute more tests. Here is a snippet from this PR's CI:
```
Run pytest -We .
============================= test session starts =============================
platform win32 -- Python 3.11.6, pytest-7.4.2, pluggy-1.3.0
rootdir: D:\a\devops_tests\devops_tests
collected 15 items

test_notebooks.py ...                                                    [ 20%]
test_todos_annotated.py ........s...                                     [100%]

======================== 14 passed, 1 skipped in 7.77s ========================
```
and here's an equivalent from the main branch (https://github.com/open-atmos/devops_tests/actions/runs/5911809701/job/16034911272):
```
Run pytest -We .
============================= test session starts =============================
platform win32 -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
rootdir: D:\a\devops_tests\devops_tests
collected 13 items

test_notebooks.py ...                                                    [ 23%]
test_todos_annotated.py .......s..                                       [100%]

======================== 12 passed, 1 skipped in 7.77s ========================
```
So somehow, `find_files` produces two more test items.... any idea? (I guess this supports the need for a unit test of `find_files`... and of course no unit test will ever save us from incorrect use of the tested code)

Comment (slayoo) [reply utils.py:9]: probably that is because we add `__init__.py` and `utils.py`?

Comment (abulenok) [reply utils.py:9]: yes, all files count as separate test cases

Comment (abulenok) [reply utils.py:9]: I think this is not a problem that we additionally include `__init__.py`

**Review by slayoo — APPROVED** [review]
